parser/tests/specs_test.rs

```diff
@@ -160,12 +161,21 @@ fn generate_actual_spans(
     input_message: &'text str,
     output: &'text mut String,
     source_text_info: &'text SourceTextInfo<'text>,
+    last_start: Location,
   }
 
   impl SpanDebuggerVisitor<'_> {
     fn print(&mut self, name: &str, span: Span) {
       assert!(name.len() <= SPAN_LABEL_WIDTH);
 
+      if span.start < self.last_start {
```

Comment:
We should also fail if `span.start < self.last_end && self.last_end < span.end`, to prevent this:
```
   ^^^
   ^^^^^
```
while allowing these two:
```
   ^^^^^
   ^^^
``````
   ^^^
      ^^
```

Reply:
This doesn't work - discussed offline